### PR TITLE
Add findtg features

### LINKED ISSUE #12: Find Students by Tutorial Group (Sorts found students by name)
As a CS TA, I can look for students by tutorial group, so that I find the appropriate students for my tutorial groups easily

### PR DESCRIPTION
**Pull request by hqhqhq1**
Add findtg features
note: list will always be sorted (in this version: change in model)
resolve #12

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -464,18 +481,31 @@ testers are expected to do more *exploratory* testing.
...
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+2. _{ more test cases …​ }_
+
+### Finding a tutorial group
+
+1. Finding students from a particular tutorial group while all students are being shown
```
Comment (vanessaxuuan) [inline docs/DeveloperGuide.md:499]: add `.`

**Comment by vanessaxuuan** [comment]
looks good, can be merged once conflict is resolved

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -394,7 +394,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. TA requests to find a tutorial group.
+2. TACH list out all the students from the tutorial group.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The tutorial group entered is not found in any student.
+    * 1a1. TACH prompts that 0 student in the list.
```
Comment (LowJiaHao99) [inline docs/DeveloperGuide.md:409]: Does 'prompts that there is 0 student in the list' sound better?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -464,18 +481,31 @@ testers are expected to do more *exploratory* testing.
 
 1. Deleting a person while all persons are being shown
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+   1. Prerequisites: List all students using the `list` command. Multiple students in the list.
 
    1. Test case: `delete 1`<br>
       Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
```
Comment (LowJiaHao99) [inline docs/DeveloperGuide.md:487]: Do you think we should use 'contact' or'student'?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -464,18 +481,31 @@ testers are expected to do more *exploratory* testing.
...
-1. _{ more test cases …​ }_
+2. _{ more test cases …​ }_
+
+### Finding a tutorial group
+
+1. Finding students from a particular tutorial group while all students are being shown
+   1. Prerequisites: Multiple students are added to a tutorial group eg:`CS2103T W15`.
+
+   2. Test case: `findtg CS2103T W15`<br>
+      Expected: all students added to the tutorial group `CS2103T W15` are listed out. Number of students list out are shown in the status message.
```
Comment (LowJiaHao99) [inline docs/DeveloperGuide.md:503]: Do you think it should be "listed out" or "list out

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -464,18 +481,31 @@ testers are expected to do more *exploratory* testing.
...
+
+   3. Test case: `findtg CS2103T`<br>
+      Expected: No student listed. Since it only provide module code without tutorial group details. All invalid entry of tutorial group or no matching tutorial group will result in no student listed.
```
Comment (LowJiaHao99) [inline docs/DeveloperGuide.md:506]: Perhaps "Since user input only module code ..." would sound better?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -137,6 +137,19 @@ Examples:
...
+Finds all students in a particular tutorial group.
+
+Format `findtg TUTORIAL_GROUP`
+
+* The search is case-insensitive. e.g. `cs2101 g08` will match `CS2101 G08`
+* Only the tutorial group is searched.
+* Only the exact tutorial group will be matched e.g. `CS2101 G` **WILL NOT** match `CS2101 G08`
+* Students with matching tutorial group will be returned. e.g. `findtg CS2101 G08` will return 
+`Charles Martinet` and `Susan Boyle` only if both of them are in the tutorial group `CS2101 G08`
+* If no students with matching tutorial group, no student will be returned (an empty list).
```
Comment (LowJiaHao99) [inline docs/UserGuide.md:151]: Perhaps "If there are no students with ..." would sound better?

**Review by LowJiaHao99 — CHANGES_REQUESTED** [review]
Mostly minor issues in Dg and ug. Once resolved, LGTM.

**Review by LowJiaHao99 — APPROVED** [review]

**Comment by LowJiaHao99** [comment]
LGTM

**Pull request merged**